packages/backend-test-utils/src/next/implementations/mockDatabaseService.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DatabaseManager } from '@backstage/backend-common';
+import {
+  coreServices,
+  createServiceFactory,
+} from '@backstage/backend-plugin-api';
+import { ConfigReader } from '@backstage/config';
+
+/** @public */
+export const mockDatabaseFactory = createServiceFactory({
+  service: coreServices.database,
+  deps: {
+    config: coreServices.config,
+    plugin: coreServices.pluginMetadata,
+  },
+  async factory({ config }) {
+    const databaseManager = config.getOptional('backend.database')
+      ? DatabaseManager.fromConfig(config)
+      : DatabaseManager.fromConfig(
+          new ConfigReader({
+            backend: {
+              database: { client: 'better-sqlite', connection: ':memory:' },
+            },
+          }),
+        );
```

Comment:
Wonder if we should just straight up make this the default behavior of the default database service factory instead?